### PR TITLE
refact: adopt all modules run actions well in toolchain & enhance the ci logic

### PR DESCRIPTION
**Pull request by imbajin**
This pr test the ci problems in #267 

TODO:
- [x] find a way to judge if the action should/need run in this pr
- [x] avoid `odjbc` influence other module building
- [x] avoid `spring-boot` dual parent problem & use it correctly
- [ ] fix hubble relative path bug & can't package be part
- [ ] use a better way to avoid `cd xx` (sub-module)
- [ ] unify more action (like build all modules together & enable checkstyle in root)
- [ ] replace dependencies in sub-modules with `org.apache.hugegraph`
- [ ] delete or remove inner bd-url in [yarn.lock](https://github.com/apache/incubator-hugegraph-toolchain/blob/master/hugegraph-hubble/yarn.lock) & check/delete all inner url (need fe)
- [ ] let sub-module use their dependency first (important)

Now we found the parent configs override the childs', it doesn't make sense (comment root config now), and we need a config to choose not use parent config

### CODE REVIEW

**Comment by z7658329** [comment]
zoulei@zouleis-MacBook-Air hubble-be % mvn license:aggregate-add-third-party
[INFO] Scanning for projects...
[INFO] 
[INFO] -------------------< org.apache.hugegraph:hubble-be >-------------------
[INFO] Building hubble-be 1.0.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- license-maven-plugin:1.19:aggregate-add-third-party (default-cli) @ hubble-be ---
[WARNING] The goal is skip due to packaging 'jar'
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.179 s
[INFO] Finished at: 2022-04-28T23:26:08+08:00
[INFO] ------------------------------------------------------------------------
zoulei@zouleis-MacBook-Air hubble-be %

File: hugegraph-loader/assembly/travis/install-mysql.sh [hunk 1]
```diff
@@ -2,22 +2,36 @@
...
 set -ev
 
-TRAVIS_DIR=`dirname $0`
+TRAVIS_DIR=$(dirname $0)
+
+# Need speed up it
+CONF=hugegraph-test/src/main/resources/hugegraph.properties
+MYSQL_USERNAME=root
+
+# Set MySQL configurations
```
Comment (javeme) [inline hugegraph-loader/assembly/travis/install-mysql.sh:11]: no content for this section?

File: .github/workflows/loader-ci.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#        with:
+#          host port: 3306 # Optional, default value is 3306. The port of host
+#          container port: 3306 # Optional, default value is 3306. The port of container
+#          character set server: 'utf8mb4' # Optional, default value is 'utf8mb4'. The '--character-set-server' option for mysqld
+#          collation server: 'utf8mb4_general_ci' # Optional, default value is 'utf8mb4_general_ci'. The '--collation-server' option for mysqld
+#          mysql version: '5.7' # Optional, default value is "latest". The version of the MySQL
+#          mysql database: ${{ env.DB_DATABASE }} # Optional, default value is "test". The specified database which will be create
+#          mysql root password: ${{ env.DB_PASS }} # Required if "mysql user" is empty, default is empty. The root superuser password
+#          #mysql user: 'root' # Required if "mysql root password" is empty, default is empty. The superuser for the specified database. Can use secrets, too
+#          #mysql password: ${{ secrets.DatabasePassword }} # Required if "mysql user" exists. The password for the "mysql user"
```
[the remark this answers is not in the document]
Comment (imbajin) [reply .github/workflows/loader-ci.yml:67]: maybe it's useful in future, but delete it anyway..

[the remark this answers is not in the document]
Comment (imbajin) [reply .github/workflows/loader-ci.yml:67]: maybe it's useful in future, but delete it anyway..

File: hugegraph-loader/assembly/travis/install-mysql.sh (hunk 1 above)
Comment (imbajin) [reply hugegraph-loader/assembly/travis/install-mysql.sh:11]: seems no need now, could remove it or keep the comment for future use

Comment (imbajin) [reply hugegraph-loader/assembly/travis/install-mysql.sh:11]: seems no need now, could remove it or keep the comment for future use

**Comment by imbajin** [comment]
> skip due to packaging 'jar'

seems it was also `jar` mode before, and what will happens if it run in `hubble` dir (not `be`)?

**Comment by javeme** [comment]
ci error:
```java
cp "${TRAVIS_DIR}"/jacocoagent.jar hugegraph-hubble-*/lib || exit 1
cp: cannot create regular file 'hugegraph-hubble-*/lib': No such file or directory
Error: Process completed with exit code 1.
```

File: hugegraph-hubble/hubble-dist/assembly/travis/run-api-test.sh [hunk 3]
```diff
@@ -2,8 +2,11 @@
 
 set -ev
 
-"$TRAVIS_DIR"/install-hugegraph-hubble.sh
+EXEC_PATH="hubble-dist/assembly/travis"
```
Comment (javeme) [inline hugegraph-hubble/hubble-dist/assembly/travis/run-api-test.sh:5]: keep "TRAVIS_DIR" style

File: hugegraph-hubble/hubble-dist/assembly/travis/install-hugegraph-hubble.sh [hunk 4]
```diff
@@ -4,9 +4,10 @@ set -ev
 
 TRAVIS_DIR=$(dirname "$0")
 
-mvn -e clean package -DskipTests
+echo "$TRAVIS_DIR"
+pwd && mvn -e -X clean package -DskipTests
 
-cp "${TRAVIS_DIR}"/jacocoagent.jar hugegraph-hubble-*/lib
-cp "${TRAVIS_DIR}"/start-hubble.sh hugegraph-hubble-*/bin
+cp "${TRAVIS_DIR}"/jacocoagent.jar hugegraph-hubble-*/lib || exit 1
```
Comment (javeme) [inline hugegraph-hubble/hubble-dist/assembly/travis/install-hugegraph-hubble.sh:10]: add `HUBBLE_DIR=$(echo hugegraph-hubble-*.*)` and use `${HUBBLE_DIR}/lib`

**Comment by z7658329** [comment]
> 

i will test

**Comment by imbajin** [comment]
separate hubble build package **problem** in another PR for ci running first

**Review by javeme — APPROVED** [review]

**Comment by z7658329** [comment]
> > skip due to packaging 'jar'
> 
> seems it was also `jar` mode before, and what will happens if it run in `hubble` dir (not `be`)?
result:
image

**Comment by imbajin** [comment]
@z7658329 the hubble build problem will fix in another pr, any problems in other modules now?

**Review by corgiboygsj — APPROVED** [review]

**Comment by z7658329** [comment]
> @z7658329 the hubble build problem will fix in another pr, any problems in other modules now?

OK

**Pull request merged**